bin/lib/installable/installable.py

```diff
@@ -122,6 +126,9 @@ def is_installed(self) -> bool:
             res_call = self.install_context.check_output(
                 self.check_call, env=self.check_env, stderr_on_stdout=self.check_stderr_on_stdout
             )
+
+            self.save_version(self.check_call[0], res_call)
```

Comment:
Couple things here:

- I think we should do any `path` manipulation here where we have the context of what we were _trying_ to run here, instead of splitting it between here and the `save_version` (all the relative_to stuff)
- this makes a somewhat surprising tie between the `check_install`'s output and the CE main repo. This is pretty spooky action at a distance; any "trunk" compiler's installation check needs to output the "version" information instead of anything else. That _seems_ to be ok for now but some compiler checks use `--help` etc

that is...with the path stuff in `save_version` I'd think we could "just" directly `stat` `self.check_call[0]` here without any special casing? What am I missing? :)

Reply:
the biggest thing is that we need to know the exact path where the compiler is installed so we can match it against a compiler that's invoked on the website

We do not have the compiler id here, just the --help check and the "actual" destination

so for example `/opt/compiler-explorer/gcc-trunk-20231008` plus `bin/g++ --help`

but to match it against the website, we need to have the symlink which is `/opt/compiler-explorer/gcc-snapshot`

that's why i just pass the `bin/g++` bit, because this function does not know about the symlink

Reply:
got it. it's sad we have this in two places, that's all. I worry about our ability to keep all these disparate ways of doing a similar thing in sync. Maybe we should delegate to the JS code (somehow?) to do this update? Or...rephrase everything a bit more? Or...plan to make a chance and accept this as is for now with all the caveats...

Reply:
it does make sense to pass the stat info here (the stat modified time is just for debugging purposes to see if everything still working)

Reply:
ok, so the path is `gcc-trunk-20231008/bin/g++`, but if I make that a full path, I'm losing the information of what the relative executable path is (`bin/g++`)